frontend/src/main/java/com/google/tripmeout/frontend/servlets/PlaceIndividualServlet.java

```diff
@@ -0,0 +1,154 @@
+package com.google.tripmeout.frontend.servlets;
+
+import com.google.gson.Gson;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.InvalidRequestException;
+import com.google.maps.model.PlaceDetails;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.PlaceVisitNotFoundException;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceIndividualServlet extends HttpServlet {
+  private final PlaceVisitStorage placeStorage;
+  private final Gson gson;
+
+  private static final Pattern URI_NAME_PATTERN =
+      Pattern.compile(".*/trips/([^/]+)/placeVisits/([^/]+)");
+
+  private final GeoApiContext CONTEXT =
+      new GeoApiContext.Builder().apiKey("AIzaSyBbXCXC2uWv3baNmirLtqUYbFsFwCXqLV8").build();
+
+  public PlaceIndividualServlet(PlaceVisitStorage placeStorage, Gson gson) {
+    this.placeStorage = placeStorage;
+    this.gson = gson;
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      Matcher matcher = ServletUtil.matchUriOrThrowError(request, URI_NAME_PATTERN);
+
+      String tripId = matcher.group(1);
+      String placeId = matcher.group(2);
+
+      Optional<PlaceVisitModel> place =
+          ServletUtil.extractFromRequestBody(request, gson, PlaceVisitModel.class);
+
+      if (!place.isPresent()) {
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+
+      PlaceVisitModel.UserMark status = place.get().userMark();
+
+      PlaceDetailsRequest placeRequest = new PlaceDetailsRequest(CONTEXT).placeId(placeId).fields(
+          PlaceDetailsRequest.FieldMask.NAME, PlaceDetailsRequest.FieldMask.GEOMETRY_LOCATION);
+      PlaceDetails details = placeRequest.await();
+
+      PlaceVisitModel newPlace = PlaceVisitModel.builder()
+                                     .setTripId(tripId)
+                                     .setPlaceId(placeId)
+                                     .setName(details.name)
```

Comment:
This line confuses me. We need to figure out what all of these parts of the PlaceVisit REST object mean.